src/pages/Balance/index.tsx

```diff
@@ -182,12 +178,20 @@ const Balance: React.FC<BalanceProps> = () => {
       }
       if (newToToken.coinGeckoId !== from.coinGeckoId)
         throw generateError(`From token ${from.coinGeckoId} is different from to token ${newToToken.coinGeckoId}`);
+
+      const latestOraiAddress = await window.Keplr.getKeplrAddr();
+      // has to switch network to the correct chain id on evm since users can swap between network tokens
+      if (!window.Metamask.isTron(from.chainId) && !from.cosmosBased) {
```

Comment:
I think its best if we directly validate if from is evm based. This condition will go wrong if there's another network that is neither tron or cosmos based (like bitcoin)